elide-core/src/main/java/com/yahoo/elide/core/FilterScope.java

```diff
@@ -26,19 +27,32 @@
     @Getter private final RequestScope requestScope;
     @Getter private final boolean isAny;
     @Getter private final List<Check<T>> checks;
+    @Getter private final List<UserCheck> userChecks;
     private UserPermission filterUserPermission = null;
 
     public FilterScope(RequestScope requestScope) {
         this.requestScope = requestScope;
         this.isAny = false;
         checks = Collections.emptyList();
+        userChecks = Collections.emptyList();
     }
 
-    public FilterScope(RequestScope requestScope, boolean isAny, Class<? extends Check>[] checkClasses) {
+    public FilterScope(RequestScope requestScope,
+                       boolean isAny,
+                       Class<? extends Check>[] checkClasses,
+                       Class<? extends UserCheck>[] userCheckClasses) {
         this.requestScope = requestScope;
         this.isAny = isAny;
 
         List<Check<T>> checks = new ArrayList<>(checkClasses.length);
+        List<UserCheck> userChecks = new ArrayList<>(userCheckClasses.length);
+        for (Class<? extends UserCheck> checkClass : userCheckClasses) {
+            try {
+                userChecks.add(checkClass.newInstance());
+            } catch (InstantiationException | IllegalAccessException e) {
+                userChecks.add(null);
```

Comment:
Log an error and maybe we should throw a runtime exception (IllegalStateException) here.